azurerm/internal/services/network/data_source_service_tags.go

```diff
@@ -0,0 +1,89 @@
+package network
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+func dataSourceServiceTags() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceServiceTagsRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": azure.SchemaLocation(),
+
+			"service": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"address_prefixes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceServiceTagsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ServiceTagsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	location := d.Get("location").(string)
+	res, err := client.List(ctx, location)
+	if err != nil {
+		return err
+	}
+
+	service := d.Get("service").(string)
+	region := d.Get("region").(string)
```

Comment:
Can we normalize this?

```suggestion
	region := azure.NormalizeLocation(d.Get("region").(string))
```